flows/routers/tests/tests.go

```diff
@@ -146,8 +147,20 @@ func HasWaitTimedOut(env utils.Environment, value types.XValue) types.XValue {
 		return types.NewXErrorf("must be called with a run as first argument")
 	}
 
-	if run.Session().Wait() != nil && run.Session().Wait().HasTimedOut() {
-		return XTestResult{true, nil}
+	// look to see if the last input event was a message or a timeout
+	runEvents := run.Events()
```

Comment:
previously this test worked by examining the wait on the session which is why we could never clear waits - now it checks events, and waits are cleared on flow resumption